pkg/cloudprovider/aws/instance.go

```diff
@@ -262,6 +327,22 @@ func (p *InstanceProvider) instanceToNode(instance *ec2.Instance, instanceTypes
 	return nil, fmt.Errorf("unrecognized instance type %s", aws.StringValue(instance.InstanceType))
 }
 
+func (p *InstanceProvider) addToIceCache(overrides *ec2.FleetLaunchTemplateOverrides, capacityType string) {
+	instanceType := aws.StringValue(overrides.InstanceType)
+	zone := aws.StringValue(overrides.AvailabilityZone)
+	cacheTTL := iceCacheSpotTTL
+	if capacityType == v1alpha1.CapacityTypeOnDemand {
+		cacheTTL = iceCacheODTTL
+	}
+	cacheKey := createIceCacheKey(instanceType, capacityType)
+	if zones, exists := p.iceCache.Get(cacheKey); exists {
+		zones.(sets.String).Insert(zone)
+		p.iceCache.Set(cacheKey, zones, cacheTTL)
```

Comment:
Thinking about this, I think there might be a weird case
- zone-a is added to the cache, 5m
- 4 minutes pass 
- zone-b is added to the cache, 5m
- both zone-a and zone-b are now cached for 5m
- 1 minute passes
- zone-a might be available again.

You could potentially store a timestamp in the cache to keep track of this, but it feels a bit complicated to me. It's probably fine in the grand scheme. Happy to discuss this more.